repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -90,6 +130,54 @@ object Storage{
       }
     }
 
+    def classFilesListSave(pkg: String,
+                           wrapper: String,
+                           dataList: List[(String, String)],
+                           imports: Imports,
+                           tag: String): Unit = {
+      val dir = pkg.replace("/", "$div") + "." + wrapper.replace("/", "$div") + "0"
+      val codeCacheDir = compileCacheDir/dir
+      if(!exists(codeCacheDir)) {
+        mkdir(codeCacheDir)
+        write(
+          codeCacheDir/classFilesOrder,
+          upickle.default.write((tag, dataList.reverse), indent = 4)
+        )
+        write(codeCacheDir/"imports.list", upickle.default.write(imports))
+      }
+    }
+
+    def classFilesListLoad(pkg: String,
+                           wrapper: String,
+                           cacheTag: String): Option[(Seq[ClassFiles], Imports)] = {
+
+      val dir = pkg.replace("/", "$div") + "." + wrapper.replace("/", "$div") + "0"
+      val codeCacheDir = compileCacheDir/dir
+
+      if(!exists(codeCacheDir)) None
+      else {
+        val metadataJson = Try {
+          read(codeCacheDir/classFilesOrder)
+        }.toOption
+        val impFile = Try {
+          read(codeCacheDir/"imports.list")
+        }.toOption
+        val (loadedTag, classFilesList) =
+          upickle.default.read[(String, Seq[(String, String)])](metadataJson.getOrElse(""))
```

Comment:
Why is there a `getOrElse("")` here? That will just cause `read` to throw and exception. What's the point?